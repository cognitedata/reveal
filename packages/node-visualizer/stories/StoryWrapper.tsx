import React from 'react';

import '@cognite/cogs.js/dist/cogs.css';

export const StoryWrapper = ({ children }) => {
  return <div style={{ flex: '1' }}>{children}</div>;
};
