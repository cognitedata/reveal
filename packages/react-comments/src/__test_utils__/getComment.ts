import { CommentResponse } from '@cognite/comment-service-types';
import { MessageType } from '@cognite/cogs.js';

export const getComment = (): CommentResponse => {
  return {
    id: 'test-id',
    _owner: 'test-_owner',
    comment: 'test-comment',
    target: { id: 'test-id', targetType: 'test-target' },
    lastUpdatedTime: '2021-08-26T07:39:12.092Z',
  };
};

// the cogs type
export const getMessage = (): MessageType => {
  return {
    id: 'test-id',
    user: 'test-user',
    text: 'test-comment',
    timestamp: 1629962411026,
  };
};
