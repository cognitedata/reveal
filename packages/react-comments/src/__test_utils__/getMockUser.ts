import {
  UMSUser,
  UMSUserProfile,
} from '@cognite/user-management-service-types';

export const getMockUser = (extras: Partial<UMSUser> = {}): UMSUserProfile => {
  return {
    id: '1',
    displayName: 'test name',
    createdTime: '',
    lastUpdatedTime: '',
    preferences: {
      hidden: false,
    },
    projects: [],
    ...extras,
  };
};
