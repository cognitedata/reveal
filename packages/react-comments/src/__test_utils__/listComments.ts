import { rest } from 'msw';
import { setupServer } from 'msw/node';

export const serviceUrl = 'https://localhost';
export const testProject = 'test-project';

const handlers = [
  rest.post(`${serviceUrl}/${testProject}/comment/list`, (_req, res, ctx) => {
    // console.log('Returning data from mock!');
    return res(
      ctx.status(200),
      ctx.json({
        items: [
          {
            id: 1,
            comment: 'first comment',
            lastUpdatedTime: '2021-08-20T08:31:24.842Z',
          },
          {
            id: 2,
            comment: 'second comment',
            lastUpdatedTime: '2021-08-20T08:32:24.842Z',
          },
        ],
      })
    );
  }),
];

const server = setupServer(...handlers);

export const mockListComments = server;
