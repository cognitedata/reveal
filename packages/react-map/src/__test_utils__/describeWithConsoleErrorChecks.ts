export const describeWithConsoleErrorChecks = (
  title: string,
  count: number,
  testCallback: () => void
) => {
  describe(`${title}`, () => {
    const origConsole = global.console;
    beforeAll(() => {
      // @ts-expect-error - missing other keys
      global.console = {
        error: jest.fn(),
        log: origConsole.log,
        warn: origConsole.warn,
      }; // used for test console errors
    });
    afterAll(() => {
      global.console = origConsole;
    });

    testCallback();

    it('should catch console.error calls', () => {
      // Error: Failed to initialize WebGL
      // eslint-disable-next-line no-console
      expect(console.error).toHaveBeenCalledTimes(count);
    });
  });
};
