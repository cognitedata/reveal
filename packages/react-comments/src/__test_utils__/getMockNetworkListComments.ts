import { rest } from 'msw';

import { getComment } from './getComment';

export const serviceUrl = 'https://localhost';
export const testProject = 'test-project';

export const getMockNetworkListComments = () =>
  rest.post(`${serviceUrl}/${testProject}/comment/list`, (_req, res, ctx) => {
    // console.log('Returning data from mock!');
    return res(
      ctx.status(200),
      ctx.json({
        items: [
          getComment({
            id: '1',
            comment: 'first comment',
          }),
          getComment({
            id: '2',
            displayName: 'test-displayName',
            comment: 'second comment',
          }),
        ],
      })
    );
  });
