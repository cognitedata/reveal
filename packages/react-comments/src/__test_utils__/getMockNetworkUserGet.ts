import { rest } from 'msw';

import { getMockUser } from './getMockUser';

export const serviceUrl = 'https://localhost';

export const getMockNetworkUserGet = (id: string) => {
  const url = `${serviceUrl}/user/${id}`;
  return rest.get(url, (_req, res, ctx) => {
    // console.log('Returning data from mock!');
    return res(ctx.status(200), ctx.json(getMockUser()));
  });
};
