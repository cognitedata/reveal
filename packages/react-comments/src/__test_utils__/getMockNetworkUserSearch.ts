import { UMSUser } from '@cognite/user-management-service-types';
import { RequestHandler, rest } from 'msw';

import { getMockUser } from './getMockUser';

type MSWRequest = RequestHandler<any, any, any, any>;

export const getMockNetworkUserSearch = (user?: Partial<UMSUser>): MSWRequest =>
  rest.post(`https://localhost/user/search`, (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json([getMockUser(user)]));
  });
