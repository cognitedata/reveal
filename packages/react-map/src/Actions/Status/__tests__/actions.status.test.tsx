import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
import { describeWithConsoleErrorChecks } from '__test_utils__/describeWithConsoleErrorChecks';

import * as stories from '../__stories__/actions.status.stories';

const testCases = Object.values(composeStories(stories)).map((Story) => [
  // The ! is necessary in Typescript only, as the property is part of a partial type
  Story.storyName!,
  Story,
]);

describeWithConsoleErrorChecks('Action Status Stories', 2, () => {
  // Batch snapshot testing
  test.each(testCases)('Renders %s story', async (_storyName, Story) => {
    const { baseElement } = await render(<Story />);
    expect(baseElement).toMatchSnapshot();
  });
});
