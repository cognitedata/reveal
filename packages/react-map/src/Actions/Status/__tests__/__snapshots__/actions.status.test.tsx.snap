// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Status Stories Renders Editing story 1`] = `
<body>
  <div>
    <div
      class="styled0 styled1"
      data-testid="shortcut-helper"
    >
      <i
        class="cogs-icon cogs-icon-Info"
        style="width: 16px;"
      >
        <svg
          fill="none"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            clip-rule="evenodd"
            d="M8 13.537A5.537 5.537 0 1 0 8 2.463a5.537 5.537 0 0 0 0 11.074ZM15 8A7 7 0 1 1 1 8a7 7 0 0 1 14 0Zm-7 3.97a.836.836 0 0 1-.836-.836V8a.836.836 0 0 1 1.672 0v3.134A.836.836 0 0 1 8 11.97Zm0-8.15a1.045 1.045 0 1 0 0 2.09 1.045 1.045 0 0 0 0-2.09Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
      <div
        class="styled2"
        data-testid="finish-info-message"
      >
        Press 
        <div
          class="styled3"
        >
          Enter
        </div>
         to finish editing
         
      </div>
      <div
        class="styled4"
        data-testid="info-message-separator"
      />
      <div
        class="styled2"
        data-testid="cancel-info-message"
      >
        Press 
        <div
          class="styled3"
        >
          Esc
        </div>
         to cancel polygon search
      </div>
    </div>
  </div>
</body>
`;

exports[`Action Status Stories Renders Exists story 1`] = `
<body>
  <div>
    <div
      class="styled0 styled1"
      data-testid="shortcut-helper"
    >
      <i
        class="cogs-icon cogs-icon-Info"
        style="width: 16px;"
      >
        <svg
          fill="none"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            clip-rule="evenodd"
            d="M8 13.537A5.537 5.537 0 1 0 8 2.463a5.537 5.537 0 0 0 0 11.074ZM15 8A7 7 0 1 1 1 8a7 7 0 0 1 14 0Zm-7 3.97a.836.836 0 0 1-.836-.836V8a.836.836 0 0 1 1.672 0v3.134A.836.836 0 0 1 8 11.97Zm0-8.15a1.045 1.045 0 1 0 0 2.09 1.045 1.045 0 0 0 0-2.09Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
      <div
        class="styled2"
        data-testid="finish-info-message"
      >
        Press 
        <div
          class="styled3"
        >
          Enter
        </div>
         to finish editing
         
      </div>
      <div
        class="styled4"
        data-testid="info-message-separator"
      />
      <div
        class="styled2"
        data-testid="cancel-info-message"
      >
        Press 
        <div
          class="styled3"
        >
          Esc
        </div>
         to cancel polygon search
      </div>
    </div>
  </div>
</body>
`;

exports[`Action Status Stories Renders Nothing story 1`] = `
<body>
  <div />
</body>
`;

exports[`Action Status Stories Renders WithMap story 1`] = `
<body>
  <div>
    <div
      class="styled0"
    >
      <div
        class="styled1 maplibregl-map mapboxgl-map"
        data-testid="map-container"
      >
        <div
          class="styled2"
        />
        <div
          class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive"
        >
          <canvas
            aria-label="Map"
            class="maplibregl-canvas mapboxgl-canvas"
            height="300"
            role="region"
            style="width: 400px; height: 300px;"
            tabindex="0"
            width="400"
          />
        </div>
        <div
          class="maplibregl-control-container mapboxgl-control-container"
        >
          <div
            class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left"
          />
          <div
            class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right"
          />
          <div
            class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left"
          />
          <div
            class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
